egui/src/containers/collapsing_header.rs

```diff
@@ -141,6 +141,7 @@ pub struct CollapsingHeader {
     selectable: bool,
     selected: bool,
     show_background: bool,
+    icon: Option<Box<dyn FnOnce(&mut Ui, f32, &Response)>>,
```

Comment:
`pub type IconPainter = Box<dyn FnOnce(&mut Ui, f32, &Response)>;` with a nice docstring would make this type simpler, and silence clippy

Reply:
I added the type alias and used it here.

I didn't use it in the `icon` function to keep the API simple (caller doesn't have to use `Box::new`). If [trait aliases](https://github.com/rust-lang/rust/issues/41517) ever get stabilized we could use that, but who knows if that will ever happen.